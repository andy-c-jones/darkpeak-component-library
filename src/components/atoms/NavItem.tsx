type NavItemProps = {
  // The label of the navigation item
  label: string;
  // The href of the navigation item
  href: string;
};

// A single navigation item component to but used in a navigation list
export const NavItem = ({ label, ...props }: NavItemProps) => {
  return (
    <li className="nav-item">
      <a className="nav-link" {...props}>
        {label}
      </a>
    </li>
  );
};
