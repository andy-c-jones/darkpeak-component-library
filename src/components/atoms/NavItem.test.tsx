/**
 * @vitest-environment jsdom
 */
import { render, cleanup, screen } from '@testing-library/react';
import { describe, expect, it, beforeEach } from 'vitest';
import { NavItem } from './NavItem';

beforeEach(() => {
  cleanup();
});

describe('NavItem', () => {
  it('renders without error', () => {
    const { asFragment } = render(<NavItem label="test" href="test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a link with the label passed in', () => {
    render(<NavItem label="some link label" href="test" />);

    let link = screen.getByRole('link');

    expect(link.innerHTML).toBe('some link label');
  });

  it('renders a link with the href passed in', () => {
    render(<NavItem label="some link label" href="test" />);

    let link = screen.getByRole('link');

    expect(link.getAttribute('href')).toBe('test');
  });
});
