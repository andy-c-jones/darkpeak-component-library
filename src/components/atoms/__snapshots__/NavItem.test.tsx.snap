// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavItem > renders without error 1`] = `
<DocumentFragment>
  <li
    class="nav-item"
  >
    <a
      class="nav-link"
      href="test"
    >
      test
    </a>
  </li>
</DocumentFragment>
`;
